### ISSUE TITLE
isInline() started to fail since v2.10.27

### ISSUE THREAD

**Issue by Nielsvanpach** [laravel-filament/filament#1909]
### Package

filament/forms

### Package Version

v2.10.27

### Laravel Version

v8.83.4

### PHP Version

PHP 8.1

### Bug description

After upgrading to v2.10.27 we got some failing code when checking if a field is inline with `isInline()`. I assume it might be related to https://github.com/laravel-filament/filament/pull/1883.

### Steps to reproduce

```php
$toggle = Toggle::make('is_admin')->inline();

$toggle->isInline();
```

or

```php
$toggle = Checkbox::make('is_admin');

$toggle->isInline();
```

```
Error : Typed property Filament\Forms\Components\Component::$container must not be accessed before initialization
```

### Relevant log output

```shell
Error : Typed property Filament\Forms\Components\Component::$container must not be accessed before initialization
 /Users/niels/dev/spatie/***/vendor/filament/forms/src/Components/Concerns/BelongsToContainer.php:21
 /Users/niels/dev/spatie/***/vendor/filament/forms/src/Components/Concerns/HasInlineLabel.php:20
 /Users/niels/dev/spatie/***/vendor/filament/forms/src/Components/Concerns/CanBeInline.php:20
 /Users/niels/dev/spatie/***/Context/CustomForms/FilamentParserTest.php:120
```

**Comment by danharrin** [comment]
Yeah, why are you calling `isInline()` like that?

**Comment by danharrin** [comment]
Yeah, this would have broken pre-v2.10.27 if a closure was used inside `inline()`.

**Comment by danharrin** [comment]
I'm just interested to see why `isInline()` was being called directly in the first place, we can recommend a more robust solution.

**Comment by ryangjchandler** [comment]
The public method broke in a patch release, but tbh it shouldn't have been working before anyway.

Those methods need to be marked as public for us to interact with them correctly, but should likely be marked as `@internal` and used at your own risk since we don't document them as part of the public API for the forms package. 

All fields need to be bound to a `ComponentContainer`, so I'd recommend taking this approach in your test. You can create a fake Livewire component for the container too.
